### PR TITLE
Clean up index file generation & fix bug

### PR DESCRIPTION
**Pull request by jaredly**
## Summary:
There was a bug where if there was one file that was the suffix of the other and both had queries (`ProfileWidgetsType.js` and `getProfileWidgetsType.js` in the case I was looking at), then it would produce an invalid index file, with multiple entries for the one that was the suffix of the other.

I also took the opportunity to clean up some of the logic in this file, as it was a little hard to follow.

Issue: XXX-XXXX

## Test plan:
checks! Also run on webapp, see that index files don't contain duplicates.

### CODE REVIEW

File: src/generateTypeFiles.js [hunk 1]
```diff
@@ -32,41 +35,33 @@ export const generateTypeFiles = (
...
         } else {
             const lines = indexContents.split('\n').map((line) => {
-                if (line.includes(path.basename(filePath))) {
+                if (line.includes('./' + path.basename(filePath))) {
                     return newLine;
                 }
                 return line;
             });
-            fs.writeFileSync(index, lines.join('\n'));
+            indexContents = lines.join('\n');
```
Comment (jeremywiebe) [inline src/generateTypeFiles.js:64]: Was just thinking about the hoops we're jumping through here.  I wonder if it'd be simpler to just have a `Set` and then just map it out to the file at the end. 🤔 

No action required.

File: src/generateTypeFiles.js [hunk 2]
```diff
@@ -75,12 +70,9 @@ export const generateTypeFiles = (
         // We write all generated files to a `__generated__` subdir to keep
         // things tidy.
         const targetFileName = `${typeName}.js`;
-        const generatedDir = path.join(path.dirname(fileName), '__generated__');
         const targetPath = path.join(generatedDir, targetFileName);
 
-        maybeCreateGeneratedDir(generatedDir);
-
-        const fileContents =
+        let fileContents =
```
Comment (jeremywiebe) [inline src/generateTypeFiles.js:75]: I think this can still be `const`

**Review by jeremywiebe — APPROVED** [review]
Nice...

File: src/generateTypeFiles.js (hunk 1 above)
Comment (jaredly) [reply src/generateTypeFiles.js:64]: true story
